indexing-service/src/test/java/org/apache/druid/indexing/common/task/batch/parallel/TombstoneHelperTest.java

```diff
@@ -115,9 +111,136 @@ public void tombstonesCreatedWhenNoDataInInputIntervalAndExistingSegments() thro
   }
 
+  @Test
+  public void tombstonesCreatedForReplaceWhenReplaceIsContainedInUsedIntervals() throws Exception
+  {
+    Interval usedInterval = Intervals.of("2020-02-01/2020-04-01");
+    Interval replaceInterval = Intervals.of("2020-03-01/2020-03-31");
+    Interval intervalToDrop = Intervals.of("2020-03-05/2020-03-07");
+    Granularity replaceGranularity = Granularities.DAY;
+
+    DataSegment existingUsedSegment =
+        DataSegment.builder()
+                   .dataSource("test")
+                   .interval(usedInterval)
+                   .version("oldVersion")
+                   .size(100)
+                   .build();
+    Assert.assertFalse(existingUsedSegment.isTombstone());
+    Mockito.when(taskActionClient.submit(any(TaskAction.class)))
+           .thenReturn(Collections.singletonList(existingUsedSegment));
+    TombstoneHelper tombstoneHelper = new TombstoneHelper(taskActionClient);
+
+    Set<Interval> tombstoneIntervals = tombstoneHelper.computeTombstoneIntervalsForReplace(
+        ImmutableList.of(replaceInterval),
+        ImmutableList.of(intervalToDrop),
+        "test",
+        replaceGranularity
+    );
+    Assert.assertEquals(
+        ImmutableSet.of(Intervals.of("2020-03-05/2020-03-06"), Intervals.of("2020-03-06/2020-03-07")),
+        tombstoneIntervals
+    );
+  }
+
+  @Test
+  public void tombstonesCreatedForReplaceWhenThereIsAGapInUsedIntervals() throws Exception
+  {
+    List<Interval> usedIntervals = ImmutableList.of(
+        Intervals.of("2020-02-01/2020-04-01"),
+        Intervals.of("2020-07-01/2020-11-01")
+    );
+    Interval replaceInterval = Intervals.of("2020-01-01/2020-12-01");
+    Interval intervalToDrop = Intervals.of("2020-03-01/2020-09-01");
+    Granularity replaceGranularity = Granularities.MONTH;
+
+    List<DataSegment> existingUsedSegments = usedIntervals.stream().map(
+        usedInterval -> DataSegment.builder()
+                                   .dataSource("test")
+                                   .interval(usedInterval)
+                                   .version("oldVersion")
+                                   .size(100)
+                                   .build()
+    ).collect(Collectors.toList());
+    Mockito.when(taskActionClient.submit(any(TaskAction.class))).thenReturn(existingUsedSegments);
+    TombstoneHelper tombstoneHelper = new TombstoneHelper(taskActionClient);
+
+    Set<Interval> tombstoneIntervals = tombstoneHelper.computeTombstoneIntervalsForReplace(
+        ImmutableList.of(replaceInterval),
+        ImmutableList.of(intervalToDrop),
+        "test",
+        replaceGranularity
+    );
+    Assert.assertEquals(
+        ImmutableSet.of(
+            Intervals.of("2020-03-01/2020-04-01"),
+            Intervals.of("2020-07-01/2020-08-01"),
+            Intervals.of("2020-08-01/2020-09-01")
+        ),
+        tombstoneIntervals
+    );
+  }
+
+  @Test
+  public void tombstonesCreatedForReplaceWhenUsedIntervalsDonotAlign() throws Exception
```

Comment:
I think these tests should either be about `tombstoneIntervals` or if they are about `tombstoneSegments`, then they should check the segments as well

Reply:
Renamed the tests to specify that we are testing intervals only (since that's the valuable and variable part in the generated segments).